From: virtudude <virtudude@protonmail.com>

--- a/arch/arm64/boot/dts/qcom/qcs8550-ayn-thor.dts
+++ b/arch/arm64/boot/dts/qcom/qcs8550-ayn-thor.dts
@@ -270,8 +270,6 @@
 
 		touchscreen-size-x = <1080>;
 		touchscreen-size-y = <1920>;
-		touchscreen-swapped-x-y;
-		touchscreen-inverted-x;
 
 		label = "top_touchscreen";
 	};
@@ -388,8 +386,6 @@
 
 		touchscreen-size-x = <1080>;
 		touchscreen-size-y = <1240>;
-		touchscreen-swapped-x-y;
-		touchscreen-inverted-x;
 
 		label = "bottom_touchscreen";
 
